pkg/cloud/services/securitygroup/securitygroups_test.go

```diff
@@ -760,7 +779,7 @@ func TestAdditionalControlPlaneSecurityGroup(t *testing.T) {
 						t.Fatalf("Expected to port %d, got %d", tc.expectedAdditionalIngresRule.ToPort, r.ToPort)
 					}
 
-					if !reflect.DeepEqual(r.SourceSecurityGroupIDs, tc.expectedAdditionalIngresRule.SourceSecurityGroupIDs) {
+					if !sets.New[string](tc.expectedAdditionalIngresRule.SourceSecurityGroupIDs...).Equal(sets.New[string](tc.expectedAdditionalIngresRule.SourceSecurityGroupIDs...)) {
```

Comment:
small nit:
why set comparison and not cmp.Equal (if you do not intend to use reflect library)?

Reply:
actually no reason :) I just felt like I could use a method provided by "sets" package since we make security groups a set in the controller logic.

Reply:
ok 🙂 just wanted to check if this has added benefit